core/message_pool.go

```diff
@@ -133,9 +169,9 @@ func (pool *MessagePool) UpdateMessagePool(ctx context.Context, store chain.Bloc
 	// Add all message from the old blocks to the message pool, so they can be mined again.
 	for _, blk := range oldBlocks {
 		for _, msg := range blk.Messages {
-			_, err = pool.addTimedMessage(&timedmessage{message: msg, addedAt: uint64(blk.Height)})
+			_, err = pool.addTimedMessage(ctx, &timedmessage{message: msg, addedAt: uint64(blk.Height)})
```

Comment:
blocking: just noticed line 226 which is not your bug but it is accessing pool.pending without holding the lock, needs to be fixed probably to pool.Pending()

Reply:
Resolved.